gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -308,74 +366,38 @@ private String getBackendFromRoutingGroup(HttpServletRequest request)
     private String getPreviousBackend(HttpServletRequest request)
     {
         String queryId = extractQueryIdIfPresent(request);
-
-        // Find query id and get url from cache
         if (!isNullOrEmpty(queryId)) {
             return routingManager.findBackendForQueryId(queryId);
         }
+        else if (!Strings.isNullOrEmpty(request.getRequestedSessionId())) {
+            //pin browser sessions to the same backend based on jsessionid, but load balance queries
+            return routingManager.findBackendForCookie(request.getRequestedSessionId().split("\\.")[0]);
+        }
         log.error("No backend found for queryId {}", queryId);
```

Comment:
update .. maybe we want to log inside the if statement instead .. or in addition?

Reply:
refactored the logic since this statement was unreachable due to a prior check